### ISSUE TITLE
XCAT Automation: Group Power on / off, Codeupdate

### ISSUE THREAD

**Issue by sivassrr** [openbmc/openbmc-test-automation#647]
Following Scenarios to be automated using XCAT

- [ ] Code update via XCAT ( both BMC / PNOR )

- [ ] Group power on / off

- [ ] XCAT / BMC Connected and leave the system idle for 2-3 days and resume some operation - (Problematic scenario in open-power environment)

- [ ] Network Switch manipulation

**Comment by gkeishin** [comment]
This tasks are already created and under epic https://github.com/openbmc/openbmc-test-automation/issues/599

**Comment by sivassrr** [comment]
This is part of the epic 599. 
There description not there.

**Comment by gkeishin** [comment]
Well we can update there but if u open with Zenhub plugin u ll see link to that epic..

**Comment by sivassrr** [comment]
yup, I was using firefox, Zenhub is broken. Let me update there. in 599. then. See it in Chrome.

**Comment by sivassrr** [comment]
https://github.com/openbmc/openbmc-test-automation/issues/256 - We suppose use this.

**Comment by gkeishin** [comment]
Yeah but that is not an epic so we can't tag to it. So keep it open , so that @prkatti1 can use it to do Manual if we couldn't complete automation on time to address all the requirement.

**Comment by jayeshmpatel** [comment]
Suggest to add few more test scenarios with xCAT:

1. After xCAT is connected to multiple/single compute node and BMC is at standby state,  issue BMC reset/reboot, then make sure xCAT can be used to power on
2.  After xCAT is connected to multiple/single compute node and BMC is in the standby state,  issue BMC reset/reboot, then make sure xCAT can be used to do code update.
3.   After xCAT is connected to  multiple/single compute node, power on the computer node, once host OS is running, issue BMC reset/reboot and make sure xCAT connection is still functional
4.   After xCAT is connected to multiple/single compute node, power on the compute node, once host OS is running, issue BMC reset/reboot and once it is at ready state, code update still works.

**Comment by sivassrr** [comment]
https://github.com/xcat2/xcat-core/issues/2818  - XCAT way of code update is getting enabled. We can go ahead and automate it. Once officially available we can test.
